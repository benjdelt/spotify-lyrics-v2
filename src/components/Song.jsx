import Error from './Error';
import '../styles/Song.css';

function Song({song, error}) {

  return (
    <section className="song">
      <header className="song-header">
        <img src={song.coverUrl} alt="cover" className="cover"/>
        <section className="song-info">
          <h2>{song.title}</h2>
          <h5>{song.artist}</h5>
          <p>{song.album}</p>
        </section>
      </header>
      {error.type === "noLyricsFound" ? (
        <Error error={error.message} />
      ) : (
        <section className="song-main">
          <p>
            {song.lyrics.map((lyric, ind) => <span key={ind} >{lyric}<br /></span>)}
          </p>
        </section>
      )}
    </section>
  );
}

export default Song;
