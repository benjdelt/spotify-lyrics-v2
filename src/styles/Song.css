.song {
  margin: 5em 0;
}

.song-header {
  display: flex;
}

.song-info h5 {
  font-size: 1em;
}

.cover {
  margin-left: .5em;;
  width: 200px;
}

.song-info {
  margin-left: 1em;
}

.song-main {
  margin-top: 2em;
  padding-left: 5em;
}

@media (max-width: 490px) {
  .song-header {
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  .song-info {
    margin: 1em;
  }

  .song-info>* {
    margin: .5em 0;
  }

  .song-main {
    margin: 0 1em;
    padding-left: 0;
  }
}