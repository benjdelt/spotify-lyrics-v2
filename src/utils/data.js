import SpotifyWebApi from 'spotify-web-api-js';

const spotifyApi = new SpotifyWebApi();


const getHashParams = (location = window.location) => {
  const q = location.hash.substring(2);

  // if the substring is non-empty, split it up based on hashes
  // return the parameters as an object
  if (q) {
    const qArray = q.split("/");
    const hashParams = {
      path: qArray[0],
      access_token: qArray[1],
      refresh_token: qArray[2],
      expires_in: qArray[3],
    };
    return hashParams;
  }
  return {};
};

const params = getHashParams();
const token = params.access_token;
if (token) {
  spotifyApi.setAccessToken(token);
}

export const getUser = async () => {
  const response = await spotifyApi.getMe();
  return {name: response.display_name, avatar: response.images[0].url};
}
