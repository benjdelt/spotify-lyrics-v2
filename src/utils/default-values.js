const errorTypes = {
  noLyricsFound: "noLyricsFound",
  noCurrentlyPlaying: "noCurrentlyPlaying",
}

const emptyUser = {
  name: "",
  avatarUrl: "",
};

const emptySong = {
  title: "",
  artist: "",
  album: "",
  coverUrl: "",
  lyrics: [],
};

const defaultLoading = {
  user: true,
  song: true,
};

export {
  errorTypes,
  emptyUser,
  emptySong,
  defaultLoading,
}
