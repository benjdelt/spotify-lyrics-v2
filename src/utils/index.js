import {
  errorTypes,
  emptyUser,
  emptySong,
  defaultLoading
} from "./default-values";

export {
  errorTypes,
  emptyUser,
  emptySong,
  defaultLoading,
}
