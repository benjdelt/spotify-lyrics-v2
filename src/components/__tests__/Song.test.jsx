import { render, cleanup } from '@testing-library/react';
import Song from '../Song';
import { emptySong } from '../../utils/';


afterEach(cleanup)

it('should take a snapshot', () => {
  const { asFragment } = render(<Song song={emptySong} />) 
  expect(asFragment(<Song song={emptySong} />)).toMatchSnapshot()
});
