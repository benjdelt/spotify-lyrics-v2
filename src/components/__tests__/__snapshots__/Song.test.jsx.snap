// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <section
    class="song"
  >
    <header
      class="song-header"
    >
      <img
        alt="cover"
        class="cover"
        src=""
      />
      <section
        class="song-info"
      >
        <h2 />
        <h5 />
        <p />
      </section>
    </header>
    <section
      class="song-main"
    >
      <p />
    </section>
  </section>
</DocumentFragment>
`;
